source/base/config.cc

```diff
@@ -25,7 +25,7 @@ namespace numbers
   bool is_finite (const double x)
   {
 #ifdef DEAL_II_HAVE_ISFINITE
-    return std::isfinite (x);
+    return !isnan(x) && std::isfinite (x);
```

Comment:
Neither isnan nor isfinite is mentioned in the C++ standard by name. In fact, the C99 standard says that they are both to be implemented as macros, so I decided to err on the safe side and go with the C99 name (for isfinite, we have already established by years of use that using `std::` seems to be ok).